src/config.rs

```diff
@@ -63,3 +108,67 @@ impl Default for Theme {
         }
     }
 }
+
+fn deserialize_map_data<'de, D>(deserializer: D) -> Result<MapData, D::Error>
```

Comment:
This is a bit hairy, I agree, but I really really wanted to use an authoring-friendly format for custom maps, instead of something cursed like:

```toml
[map.data]
0 = [ "wall", "wall", "empty", ...]
1 = ...
```

So, I followed the [Serde docs](https://serde.rs/impl-deserializer.html) on how to create a custom deserialization for a type. In this case I could just make this method instead using the decorator you see in line 57.

Basically we write this dummy struct that implements `visitor`. It has 1 required method, `expecting`, for the syntax error message, and optional `visit_{type}` methods. In this case I wanted to build the type from a string, so I implemented `visit_str`, and call `deserializer.deserialize_str` with the visitor. It is a bit clunky for something so simple but it kind of makes sense for more general visitors.

In the visit method, I split the input into lines and collect the characters into a `MapData`, which later will be used to spawn the actual entities.